File: BitManipulation/7. 整数反转/7. 整数反转.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        last = 0
        p_or_m = True if x > 0 else False
        x = abs(x)
        while (x != 0):
            tmp = x % 10 #取尾數
            res = res * 10 + tmp #答案為上次答案*10 + 此次尾數
            x = x // 10

        if(res > 2147483647 or res < -2147483648):
            return 0

        if p_or_m: 
            return res
        else:
            return -res

File: BitManipulation/7. 整数反转/7. 整数反转.cpp
class Solution {
public:
    int reverse(int x) {
        bool p_or_n = x > 0 ? true : false;
        long long res = 0; //避免x溢出

        // x不需處理絕對值問題
        while(x != 0){
            int tmp = x % 10;
            res = res*10 + tmp;
            x = x/10;
        }

        if(res > 2147483647 || res < -2147483648){
            return 0;
        }

        return res;
    }
};